arbnode/parent/parent.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2025, Offchain Labs, Inc.
+// For license information, see:
+// https://github.com/offchainlabs/nitro/blob/master/LICENSE.md
+
+package parent
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/consensus/misc/eip4844"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/offchainlabs/nitro/util/headerreader"
+)
+
+var (
+	knownConfigs = map[uint64]*params.ChainConfig{
+		params.MainnetChainConfig.ChainID.Uint64():         params.MainnetChainConfig,
+		params.HoleskyChainConfig.ChainID.Uint64():         params.HoleskyChainConfig,
+		params.SepoliaChainConfig.ChainID.Uint64():         params.SepoliaChainConfig,
+		params.AllDevChainProtocolChanges.ChainID.Uint64(): params.AllDevChainProtocolChanges,
+	}
+)
+
+type ParentChain struct {
+	ChainID  *big.Int
+	L1Reader *headerreader.HeaderReader
+}
+
+// ErrUnknownChain is returned when an unknown chain ID is requested.
+type ErrUnknownChain struct {
+	ChainID *big.Int
+}
+
+// Error implements the error interface.
+func (e ErrUnknownChain) Error() string {
+	return "unknown chain ID " + e.ChainID.String()
+}
+
+// chainConfig returns the parent chain's configuration.
+//
+// Note: This is really a hack. This method returns one of the known Ethereum
+// L1 chain configurations, but should not be used by chains that need to post
+// blobs to some other parent chain.
+func (p *ParentChain) chainConfig() (*params.ChainConfig, error) {
+	cfg, ok := knownConfigs[p.ChainID.Uint64()]
+	if !ok {
+		return nil, ErrUnknownChain{p.ChainID}
+	}
+	return cfg, nil
+}
+
+// MaxBlobGasPerBlock returns the maximum blob gas per block according to
+// according to the configuration of the parent chain.
+// Passing in a nil header will use the time from the latest header.
+func (p *ParentChain) MaxBlobGasPerBlock(ctx context.Context, h *types.Header) (uint64, error) {
+	header := h
```

Comment:
I wish there was a nice way to check if the header is from the expected chain so callers don't mix up parent and child chain headers. But alas there is no chainid on the Header type.